dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AuthenticatedResourceInterface.java

```diff
@@ -15,266 +15,181 @@
  */
 package io.dockstore.webservice.resources;
 
-import com.github.zafarkhaja.semver.Version;
-import com.google.common.collect.Lists;
 import io.dockstore.webservice.CustomWebApplicationException;
 import io.dockstore.webservice.core.Entry;
 import io.dockstore.webservice.core.Organization;
 import io.dockstore.webservice.core.Token;
 import io.dockstore.webservice.core.User;
 import java.util.List;
 import java.util.Optional;
-import javax.ws.rs.container.ContainerRequestContext;
 import org.apache.http.HttpStatus;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 /**
- * Endpoints that use authentication by Dockstore user
+ * AuthenticatedResourceInterface is a mixin that provides methods used to implement user-level access control in
+ * Resource handlers.  By using these methods consistently, we centralize access-checking logic and avoid its
+ * repetition, allowing us to easily modify our access policies, if necessary.
+ *
+ * Each of the "check" methods returns on success, and throws an appropriate CustomWebApplicationException on failure.
+ * The most commonly-used "check" methods check if a user is allowed to perform a type of action on a specified entry:
+ *
+ * <ul>
```

Comment:
might be worth it just to get some of this documentation in place. 
I'm a little surprised that we need all of these

Reply:
> I'm a little surprised that we need all of these

I tend to agree, it's a little hard to follow; for example, both `canRead` and `checkCanRead` can be overridden (although in practice, it doesn't look like the `check..` methods ever are). It would be nice if the `check...` methods were `final`, but you can't do that in an interface.

You could fold `canRead` into `checkCanRead`, etc., but everything's a tradeoff.